proposals/0296-increase-tx-size.md

```diff
@@ -0,0 +1,243 @@
+client as well as a frictional developer experience for many developers.
+
+A number of developers have also resorted to using Jito bundles to artificially
+increase the size of their transactions. This is a current workaround that
+developers are using, but it does not have the same guarantees as a transaction
+does on the protocol level, namely that bundles are not guaranteed to be atomic
+like singular transactions.
+
+For these reasons, increasing the current transaction size limit would enable
+developers to create new applications and use cases that were previously
+unfeasible.
+
+## New Terminology
+
+- 'v1 transaction' - A new transaction format that is designed to enable larger
+  transactions sizes while not having the address lookup table features 
+  introduced in v0 transactions.
+
+## Detailed Design
+
+Increasing the transaction size limit above the current MTU max size is
+possible with the use of QUIC. QUIC's RFC 9000 specification does not have an
+explicit maximum stream size, allowing for larger transactions to be sent.
+
+A new transaction format, `v1`, is proposed to enable larger transaction sizes.
+The `v1` transaction format would be:
+
+```
+VersionByte (u8) - =129 to distinguish from legacy/v0 formats
+LegacyHeader (u8, u8, u8) -- Required signatures from the current
+`MessageHeader` type
+NumInstructions (u8)
+TransactionConfigMask (u32) -- Bitmask of which config requests are present.
+LifetimeSpecificier [u8; 32]
+NumAddresses (u8)
+Addresses [[u8; 32]] -- Length matches NumAddresses
+ConfigRequests [u8] -- Array of request values. (section size is popcount 
+  TransactionConfigMask bytes). See section TransactionConfigMask for details.
+Ixs [(u8, u8, u16)] -- Number matches NumInstructions. Values are 
+ (program_account_index, num_accounts, num_data_bytes)
+[TRAILING DATA SECTION] -- Length must be consistent with the sum of 
+ num_accounts and num_data_bytes in `Ixs`
+Signatures [[u8; 64]] -- Length of `num_required_signatures` from
+ `LegacyHeader`
+```
+
+The [TRAILING DATA SECTION] contains instruction's serialized account indexes
+and instruction data bytes,
+in the order they appear in the transaction. Each instruction's account indexes
+must appear before its data bytes.
+For example, if there are two instructions:
+
+1. `Instruction { account_indexes: [0, 1], data: [2, 3, 4] }`
+2. `Instruction { account_indexes: [5, 6], data: [7, 8] }`
+
+The TRAILING DATA SECTION must look like this:
+
+```
+[0, 1, 2, 3, 4, 5, 6, 7, 8]
+```
+
+Lengths are serialized in the earlier `Ixs` section, and so are not repeated in
+the trailing data section.
+
+This new `v1` transaction format notably does not include address lookup
+tables.
+
+With the new transaction size limit, it is possible to go past some other 
+implicit limitations that are currently in place on the network. To avoid
+causing unintended issues, the following limitations are proposed to be in
+place:
```

Comment:
I think the sanitization details and table deserve a new subheading of something like `Transaction Protocol Constraints`

The details here don't need to explain why we're doing this btw, it just needs to be declarative about how the validator should be implemented. So I would move your explanation about these constraints up under the new header and before the table since this is what an implementor needs to know.

> Any transaction violating these constraints will be considered invalid and will not be included in the chain. Violations are considered sanitization failures.

Reply:
Updated and moved around the section